src/explore-education-statistics-common/src/components/form/FormBaseTextArea.tsx

```diff
@@ -0,0 +1,90 @@
+import ErrorMessage from '@common/components/ErrorMessage';
+import FormLabel, { FormLabelProps } from '@common/components/form/FormLabel';
+import classNames from 'classnames';
+import React, {
+  ChangeEventHandler,
+  FocusEventHandler,
+  KeyboardEventHandler,
+  MouseEventHandler,
+  ReactNode,
+  Ref,
+  memo,
+} from 'react';
+
+export interface FormTextAreaProps
+  extends Pick<FormLabelProps, 'hideLabel' | 'label'> {
+  className?: string;
+  disabled?: boolean;
+  error?: ReactNode | string;
+  hasMaxLengthError?: boolean;
+  hint?: string;
+  id: string;
+  inputRef?: Ref<HTMLTextAreaElement>;
+  maxLength?: number;
+  name: string;
+  rows?: number;
+  value?: string;
+  onBlur?: FocusEventHandler<HTMLTextAreaElement>;
+  onChange?: ChangeEventHandler<HTMLTextAreaElement>;
+  onClick?: MouseEventHandler<HTMLTextAreaElement>;
+  onKeyPress?: KeyboardEventHandler<HTMLTextAreaElement>;
+}
+
+function FormBaseTextArea({
+  className,
+  disabled,
+  error,
+  hasMaxLengthError = false,
+  hint,
+  id,
+  inputRef,
+  hideLabel,
+  label,
+  maxLength,
+  name,
+  rows = 5,
+  value,
+  onBlur,
+  onChange,
+  onClick,
+  onKeyPress,
+}: FormTextAreaProps) {
+  return (
+    <>
+      <FormLabel id={id} label={label} hideLabel={hideLabel} />
+
+      {hint && (
+        <div id={`${id}-hint`} className="govuk-hint">
+          {hint}
+        </div>
+      )}
+
+      {error && <ErrorMessage id={`${id}-error`}>{error}</ErrorMessage>}
+
+      <textarea
+        aria-describedby={
+          classNames({
+            [`${id}-error`]: !!error,
+            [`${id}-hint`]: !!hint,
+            [`${id}-info`]: !!maxLength,
+          }) || undefined
+        }
+        className={classNames('govuk-textarea', className, {
+          'govuk-js-character-count govuk-textarea--error': hasMaxLengthError,
```

Comment:
Doesn't seem like we actually need the `govuk-js-character-count` class? If this is the case, we don't really need the `hasMaxLengthError` prop

The current code also doesn't seem right as we need to apply `govuk-textarea--error` if `error` is truthy as well.

Reply:
yeah good spot, that's sorted now.